docs/README.md

```diff
@@ -0,0 +1,217 @@
+    MEDIAWIKI_OAUTH_CLIENT_SECRET=<your-consumer-secret>
+    MEDIAWIKI_OAUTH_CALLBACK_URL="${APP_URL}/auth/callback"
+    MEDIAWIKI_OAUTH_BASE_URL=https://www.wikidata.org
+
+    #...
+    ```
+
+1. As soon as you receive the email from the WMF team that your consumer is approved, you may start testing your application by logging in through your local instance's home page.
+
+## Troubleshooting <a id="troubleshooting"></a>
+
+### Address already in use
+
+#### MariaDB
+
+**Problem:** Sail refuses to start with the following error.
+
+```
+ERROR: for wikidata-mismatch-finder_mariadb_1  Cannot start service mariadb: driver failed programming external connectivity on endpoint wikidata-mismatch-finder_mariadb_1 (dbd1f278da0a5edb416b7875f224b83d1f2c08feac6e9a31d01d28567b83b4c7): Error starting userland proxy: listen tcp4 0.0.0.0:3306: bind: address already in use
+```
+
+**Possible Explanation:** This error most probably occurs when you already have a local system-wide instance of MySQL or MariaDB running on the default port `3306`.
+
+**Solution:** Set the port to something other than `3306` using the `FORWARD_DB_PORT` environment variable.
+
+***While running the app:***
+
+```bash
+FORWARD_DB_PORT=3308 sail up
+```
+
+OR ***In your `.env` file:***
+
+```bash
+##...
+APP_DEBUG=true
+APP_URL=http://localhost
+
+FORWARD_DB_PORT=3308
+##...
+```
+
+#### `laravel.test`
+
+**Problem:** Sail refuses to start with the following error.
+
+```
+ERROR: for wikidata-mismatch-finder_laravel.test_1  Cannot start service laravel.test: driver failed programming external connectivity on endpoint wikidata-mismatch-finder_laravel.test_1 (0fa137072f412614077154ca7927cbb0ca2f3df5474879bb5e8e33f15a1683e3): Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use
+```
+
+**Possible Explanation:** This error most probably occurs when you already have a local web server running on the default port `80`.
+
+**Solution:** Set the port to something other than `80` using the `APP_PORT` environment variable.
+
+***While running the app:***
+
+```bash
+APP_PORT=1337 sail up
+```
+
+OR ***In your `.env` file:***
+
+```bash
+##...
+APP_DEBUG=true
+APP_URL=http://localhost:1337
+
+APP_PORT=1337
+##...
+```
+
+### OAuth Error retrieving temporary credentials
+
+**Problem:** The local server returns this error on clicking the log in link.
+
+```
+League\OAuth1\Client\Credentials\CredentialsException
+Error in retrieving temporary credentials.
+http://localhost/auth/login
+```
+**Possible Explanation:**
+
+After following the steps in the [Working with OAuth section](#working-with-oauth) you might get this error. It means your OAuth Consumer Registration is still being processed and you have to wait for the consumer key and consumer secret to be approved.
+
+This error might be caused as well if you have entered the wrong consumer credentials, or a wrong APP_URL. 
+
+**Solution:**  Wait a few hours and try again.
```

Comment:
```suggestion
**Solution:**
1. Make sure your app key and secert are correctly set to your consumer key and secert in the `.env file`
1. Make sure that your `APP_URL` is set correctly to your localhost address and the port your application is running on (if running on a non default port)
1. Double check to see that the WMF has sent an email to approve your localhost oauth consumer, if they have not responses yet, please wait patiently, as the review process might take a few hours.
```

Reply:
spelling mistake: secret instead of secert

Reply:
I think I might've fixed it